lib/adviser_list_csv.rb

```diff
@@ -5,11 +5,33 @@ def initialize(advisers)
 
   def to_csv(_ = {})
     CSV.generate do |csv|
-      csv << ['Ref. Number', 'Name', 'Firm']
-
+      csv << headings
       @advisers.each do |adviser|
-        csv << [adviser.reference_number, adviser.name, adviser.firm.registered_name]
+        csv << row(adviser)
       end
     end
   end
+
+  private
+
+  def headings
+    ['Ref. Number', 'Name', 'Firm'] + all_certifications
+  end
+
+  def all_certifications
+    @all_certifications ||= Accreditation.all.pluck(:name) + Qualification.all.pluck(:name)
```

Comment:
Minor detail. Can you remove the all from both calls? Just `Accreditation.pluck(:name)` and `Qualification.pluck(:name)` will work fine.

Reply:
👍

Reply:
please address the line length